src/stackdriver-spinner/main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"os"
+
+	"log"
+
+	"time"
+
+	"strconv"
+
+	"fmt"
+	"net/http"
+
+	"github.com/cloudfoundry-community/stackdriver-tools/src/stackdriver-spinner/cloudfoundry"
+	"github.com/cloudfoundry-community/stackdriver-tools/src/stackdriver-spinner/session"
+	"github.com/cloudfoundry-community/stackdriver-tools/src/stackdriver-spinner/stackdriver"
+)
+
+func main() {
+
+	count, err := strconv.Atoi(os.Getenv("SPINNER_COUNT"))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	wait, err := strconv.Atoi(os.Getenv("SPINNER_WAIT"))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	gcpProj := os.Getenv("GCP_PROJECT")
+	if len(gcpProj) == 0 {
+		log.Fatal("A GCP project must be specified.")
+	}
+
+	go startSpinner(gcpProj, count, wait)
+
+	http.HandleFunc("/", func(res http.ResponseWriter, req *http.Request) {
+		fmt.Fprintf(res, "Johny 5 alive!")
+	})
+	fmt.Println("listening...")
+
+	err = http.ListenAndServe(":"+os.Getenv("PORT"), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func startSpinner(proj string, count, wait int) {
+	waitTime := time.Duration(wait) * time.Second
+
+	emitter := cloudfoundry.NewEmitter(os.Stdout)
+	probe, err := stackdriver.NewLoggingProbe(proj)
+	if err != nil {
+		log.Fatal(err)
+	}
+	s := session.NewSession(emitter, probe)
+	for {
+		result, err := s.Run(count, waitTime)
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+	logger, err := stackdriver.NewLogger(proj)
+
+	logger.Publish(fmt.Sprintf("GUID: %s - Found: %d - Loss: %.2f \n", result.GUID, result.Found, result.Loss))
```

Comment:
great place to start! do you plan on following this up with emitting this to stackdriver monitoring or other thoughts around that?

Reply:
The thought here was that we can emit these metrics as regular logs and then use log based metrics for monitoring and alerting. What are your thoughts on this? Would it be better to emit monitoring metrics instead?

- Sahil & Garima

Reply:
Is it easy to extract the numeric value from log based metrics and graph on that? The final use-case I see here is a graph of '% Logging Loss over time' with alerting thresholds setup by the user (eg, over 10% for 5 minutes).

I think it's good to do both because it's possible to use the Stackdriver Nozzle w/o Stackdriver Monitoring and there is associated cost with using Monitoring/custom metrics.